Konnected Pro Board support and config flow

kit-klein opened the pull request:
**Description:**
This PR tweaks the documentation to include pro board configuration parameters (extended IO) and mention the availability of config flow based config.

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#30894

## Checklist:

- [ ] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [ ] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

kit-klein committed:
Konnected Pro Board support and config flow

Updated docs to reflect pro board support and config flow support

MartinHjelmare commented on source/_integrations/konnected.markdown:
```diff
@@ -32,7 +32,11 @@ Konnected devices communicate with Home Assistant over your local LAN -- there i
-A `konnected` section must be present in the `configuration.yaml` file that specifies the Konnected devices on the network and the sensors or actuators attached to them:
+Home Assistant offers Konnected integration through **Configuration** -> **Integrations** -> **Konnected.io**.
+
+The config flow will guide you through a setup process that lets you configure the behavior of each zone and generate a config entry. 
```
Please check and update this section after the changes in the parent PR regarding the config and options flow.

kit-klein replied:
on it

kit-klein committed:
Updated to reflect options flow

MartinHjelmare reviewed (dismissed):
Looks good!

I'll leave this for the docs team to have a look too.

MartinHjelmare commented:
Please solve the merge conflict.

kit-klein committed:
Merge branch 'next' into patch-1

kit-klein committed:
Note on imports

Note that the config/option flow must be used to finalize an imported configuration.yaml.

kit-klein committed:
fix textlint

frenck commented on source/_integrations/konnected.markdown:
```diff
@@ -22,7 +22,7 @@ The integration currently supports the following device types in Home Assistant:
 - Sensor: Periodic measurements from DHT temperature/humidity sensors and DS18B20 temperature sensors.
 
-This integration uses the [`discovery`](/integrations/discovery) component, which must be enabled for device discovery to work. If you don't want to use discovery, set the _host_ and _port_ for each device in the description.  
+This integration uses the [`ssdp`](/integrations/ssdp) component, which must be enabled for device discovery to work. If you don't want to use discovery, set the _host_ and _port_ for each device in the description.  
```
Does the user have to enable SSDP that themselves? (Sound a bit weird to me).
```suggestion
This integration uses the [SSDP](/integrations/ssdp) integration, which must be enabled for device discovery to work. If you don't want to use discovery, set the _host_ and _port_ for each device in the description.
```

kit-klein replied:
ssdp is part of the default config so a user would need to manually disable it to cause issues with discovery.

frenck replied:
Now I'm even confused more.
Your last response:

> ssdp is part of the default config so a user would need to manually disable it to cause issues with discovery

The documentation:

> This integration uses the [`ssdp`](/integrations/ssdp) component, which must be enabled for device discovery to work.

So should they enable or disable it? But not enable, because it is part of the default configuration? But have to disable it for issues? what issues? How can they disable it?

🤷‍♂

kit-klein replied:
The language is attempting to notify a user that the Konnected integration requires the SSDP component to be enabled for discovery features to work.  

The SSDP component is enabled in the default home assistant config as described [here](https://www.home-assistant.io/integrations/ssdp/).   

If for some reason a User disabled SSDP, or removed the default config, they would need to enable the SSDP component for discovery to work in the Konnected integration. 

SSDP disabled seems like an exception but highlighting the dependency is still valuable.  Perhaps there is a clearer way to phrase this information?

frenck replied:
I think it is fine to assume the default and just remove that sentence. There are many other things that cause discovery not to work at all (missing mDNS repeaters on VLANs, Wi-Fi isolations or broadcast blocking, just a bad router).


```
This integration automatically discovers your Konnected devices. If you don't want to use discovery, set the `host` and `port` for each device in the description.
```

frenck commented on source/_integrations/konnected.markdown:
```diff
@@ -32,7 +32,13 @@ Konnected devices communicate with Home Assistant over your local LAN -- there i
+
+The configuration flow will guide you through a setup process that lets you connect to the panel and generate a configuration entry.  You can then utilize the options flow to configure or modify the behavior of each zone. 
+
+If you prefer you can also utilize a `konnected` section in the `configuration.yaml` file that specifies the Konnected devices on the network and the sensors or actuators attached to them.  If using `configuration.yaml` the configuration will automatically be imported and used to start a configuration flow.  **Note that you must still complete the configuration/options flow before the configurati ...
```
```suggestion
If you prefer you can also utilize a `konnected` section in the `configuration.yaml` file that specifies the Konnected devices on the network and the sensors or actuators attached to them.  If using `configuration.yaml` the configuration will automatically be imported and used to start a configuration flow. **Note that you must still complete the configuration/options flow before the configuration entry will be finalized.**
```

frenck commented on source/_integrations/konnected.markdown:
```diff
@@ -69,7 +75,7 @@ devices:
   keys:
     id:
-      description: The MAC address of the NodeMCU Wi-Fi module with colons/punctuation removed, for example `68c63a8bcd53`. You can usually find the mac address in your router's client list. Or, check the home-assistant.log for log messages from automatically discovered devices.
+      description: The MAC address of the Konnected device with colons/punctuation removed, for example `68c63a8bcd53`. You can usually find the mac address in your router's client list. Or, check the home-assistant.log for log messages from automatically discovered devices.
```
```suggestion
      description: The MAC address of the Konnected device with colons/punctuation removed, for example, `68c63a8bcd53`. You can usually find the mac address in your router's client list. Or, check the `home-assistant.log` for log messages from automatically discovered devices.
```

frenck commented on source/_integrations/konnected.markdown:
```diff
@@ -170,7 +176,10 @@ devices:
 ### Configuration Notes
 
 - Either **pin** or **zone** is required for each actuator or sensor. Do not use both in the same definition.
```
```suggestion
- Either `pin` or `zone` is required for each actuator or sensor. Do not use both in the same definition.
```

frenck commented on source/_integrations/konnected.markdown:
```diff
@@ -170,7 +176,10 @@ devices:
 ### Configuration Notes
 
 - Either **pin** or **zone** is required for each actuator or sensor. Do not use both in the same definition.
+- **pin** represents the number corresponding to the _IO index_ of the labeled pin on the NodeMCU dev board. See the [NodeMCU GPIO documentation](https://nodemcu.readthedocs.io/en/master/en/modules/gpio/) for more details. Valid values are `1`, `2`, `5`, `6`, `7`, `8`, and `9`.  Pin based configuration is only allowed with esp8266 based devices.
```
```suggestion
- `pin` represents the number corresponding to the _IO index_ of the labeled pin on the NodeMCU dev board. See the [NodeMCU GPIO documentation](https://nodemcu.readthedocs.io/en/master/en/modules/gpio/) for more details. Valid values are `1`, `2`, `5`, `6`, `7`, `8`, and `9`. Pin based configuration is only allowed with ESP8266 based devices.
```

frenck commented on source/_integrations/konnected.markdown:
```diff
@@ -170,7 +176,10 @@ devices:
 - Either **pin** or **zone** is required for each actuator or sensor. Do not use both in the same definition.
+- **pin** represents the number corresponding to the _IO index_ of the labeled pin on the NodeMCU dev board. See the [NodeMCU GPIO documentation](https://nodemcu.readthedocs.io/en/master/en/modules/gpio/) for more details. Valid values are `1`, `2`, `5`, `6`, `7`, `8`, and `9`.  Pin based configuration is only allowed with esp8266 based devices.
 - Pin `D8` or the `out` zone will only work when activation is set to high (the default).
+- **zone** represents the value corresponding to the labeled zone on the [Konnected Alarm Panel](https://konnected.io) boards. Valid zone values are `1`, `2`, `3`, `4`, `5`, `6`, and `out` for the Konnected Alarm Panel (`out` represents the dedicated ALARM/OUT terminal) and `1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `10`, `11`, `12`, `out1`, `alarm1`, and `alarm2_out2` for the Konnected Alarm P ...
```
```suggestion
- `zone` represents the value corresponding to the labeled zone on the [Konnected Alarm Panel](https://konnected.io) boards. Valid zone values are `1`, `2`, `3`, `4`, `5`, `6`, and `out` for the Konnected Alarm Panel (`out` represents the dedicated ALARM/OUT terminal) and `1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `10`, `11`, `12`, `out1`, `alarm1`, and `alarm2_out2` for the Konnected Alarm Panel Pro.
```

frenck commented on source/_integrations/konnected.markdown:
```diff
@@ -170,7 +176,10 @@ devices:
+- **pin** represents the number corresponding to the _IO index_ of the labeled pin on the NodeMCU dev board. See the [NodeMCU GPIO documentation](https://nodemcu.readthedocs.io/en/master/en/modules/gpio/) for more details. Valid values are `1`, `2`, `5`, `6`, `7`, `8`, and `9`.  Pin based configuration is only allowed with esp8266 based devices.
 - Pin `D8` or the `out` zone will only work when activation is set to high (the default).
+- **zone** represents the value corresponding to the labeled zone on the [Konnected Alarm Panel](https://konnected.io) boards. Valid zone values are `1`, `2`, `3`, `4`, `5`, `6`, and `out` for the Konnected Alarm Panel (`out` represents the dedicated ALARM/OUT terminal) and `1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `10`, `11`, `12`, `out1`, `alarm1`, and `alarm2_out2` for the Konnected Alarm P ...
+- **The Konnected Alarm Panel Pro does not support configuration via pin.**
```
```suggestion
- **The Konnected Alarm Panel Pro does not support configuration via `pin`.**
```

kit-klein committed:
Update source/_integrations/konnected.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

kit-klein committed:
Update source/_integrations/konnected.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

kit-klein committed:
Update source/_integrations/konnected.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

kit-klein committed:
Update source/_integrations/konnected.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

kit-klein committed:
Update source/_integrations/konnected.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

kit-klein committed:
Update source/_integrations/konnected.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

kit-klein committed:
Update source/_integrations/konnected.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

frenck reviewed (approved):
I'm going ahead and merge this. The comment I left is food for thought and not really blocking. 👍 
Thanks, @kit-klein, nice work on this! 👍

kit-klein commented:
@frenck Thanks for the good feedback. I totally agree with your proposal and will open another PR with the suggested line removed.
